app/models/book.rb

```diff
@@ -1,3 +1,60 @@
 class Book < ApplicationRecord
   has_and_belongs_to_many :tags
+  has_one :reservation, dependent: :destroy
+  belongs_to :user, foreign_key: "user_id"
+
+  validate :cannot_edit_reserved_or_booked
```

Comment:
I think this naming is a bit confusing too, especially because of the negation, could we change it to something like `check_status_for_editing` or similar.. Also should this live in a `before_update` callback instead?

Reply:
Both very valid points, I've updated the names of methods and changed `validate` to `before_update` (haven't used callbacks before, so yay to learning new things! 🎉 )